angular-ngrx-scss/src/app/user/services/user.service.spec.ts

```diff
@@ -55,4 +62,182 @@ describe('UserService', () => {
+          organizations_url: 'https://api.github.com/users/thisdot/orgs',
+          received_events_url:
+            'https://api.github.com/users/thisdot/received_events',
+          repos_url: 'https://api.github.com/users/thisdot/repos',
+          site_admin: false,
+          starred_url:
+            'https://api.github.com/users/thisdot/starred{/owner}{/repo}',
+          subscriptions_url:
+            'https://api.github.com/users/thisdot/subscriptions',
+          type: 'Organization',
+          url: 'https://api.github.com/users/thisdot',
+        },
+      },
+      {
+        name: 'Repo-test-3',
+        description: 'This is a repo test 2',
+        language: 'Javascript',
+        private: false,
+        stargazers_count: 0,
+        forks_count: 0,
+        updated_at: '2022-06-17T09:54:38Z',
+        owner: {
+          avatar_url: 'https://avatars.githubusercontent.com/u/22839396?v=4',
+          events_url: 'https://api.github.com/users/thisdot/events{/privacy}',
+          followers_url: 'https://api.github.com/users/thisdot/followers',
+          following_url:
+            'https://api.github.com/users/thisdot/following{/other_user}',
+          gists_url: 'https://api.github.com/users/thisdot/gists{/gist_id}',
+          gravatar_id: '',
+          html_url: 'https://github.com/thisdot',
+          id: 22839396,
+          login: 'thisdot',
+          node_id: 'MDEyOk9yZ2FuaXphdGlvbjIyODM5Mzk2',
+          organizations_url: 'https://api.github.com/users/thisdot/orgs',
+          received_events_url:
+            'https://api.github.com/users/thisdot/received_events',
+          repos_url: 'https://api.github.com/users/thisdot/repos',
+          site_admin: false,
+          starred_url:
+            'https://api.github.com/users/thisdot/starred{/owner}{/repo}',
+          subscriptions_url:
+            'https://api.github.com/users/thisdot/subscriptions',
+          type: 'Organization',
+          url: 'https://api.github.com/users/thisdot',
+        },
+      },
+    ];
+    httpClientSpy.get.and.returnValue(of(expectedHttpResponse).pipe(delay(0)));
+
+    userService.getUserTopRepos().subscribe((res) => {
+      expect(res).toEqual(expectedResponse);
+      done();
+    });
+    expect(httpClientSpy.get.calls.count()).withContext('called once').toBe(1);
+  });
+
+  it('should return the gists from the user', (done) => {
+    const username = 'thisDot';
+    const expectedResponse: UserGistsState[] = [
+      {
+        url: 'github.com/gists',
+        fileName: 'textfile1.txt',
+      },
+    ];
+
+    const expectedHttpResponse: Partial<UserGist>[] = [
+      {
+        html_url: 'github.com/gists',
+        files: { 'textfile1.txt': { filename: 'textfile1.txt' } },
+      },
+    ];
+
+    httpClientSpy.get.and.returnValue(of(expectedHttpResponse).pipe(delay(0)));
+
+    userService.getUserGists(username).subscribe((res) => {
+      expect(res).toEqual(expectedResponse);
+      done();
```

Comment:
nitpick: We should call `done()` inside of the `complete` callback of the subscription instead of the success one. If an error happens in the observable, this test may hang and never complete.

Reply:
I haven't tested this though. Maybe subscriptions work differently in tests.

Reply:
Completely agree with you @Reshurum, doing a little bit of research found this quite interesting article who mention we should avoid the `done()` method on the spec. because can cause some false positives and a slow performance on tests:

[https://www.piotrl.net/angular-testing-avoid-done/](https://www.piotrl.net/angular-testing-avoid-done/)

So basically I rewrite using `fakeAsync` to handle the Observables.